cli/flox/src/commands/environment.rs

```diff
@@ -1437,6 +1502,68 @@ impl Pull {
     }
 }
 
+fn query_add_system(system: &str) -> bool {
+    let message = format!(
+        "The environment you are trying to pull is not compatible with your system ({system})."
+    );
+    let help = "Use 'flox pull --add-system' to automatically add your system to the list of compatible systems";
+    let confirm_choice =
+        format!("Pull this environment anyway and add '{system}' to the supported systems list.");
+
+    if !Dialog::can_prompt() {
+        return false;
+    }
+
+    let dialog = Dialog {
+        message: &message,
+        help_message: Some(help),
+        typed: Select {
+            options: ["Don't pull this environment", &confirm_choice].to_vec(),
+        },
+    };
+
+    let (choice, _) = dialog.raw_prompt().unwrap_or((0, ""));
+
+    choice == 1
+}
+
+fn amend_current_system(env: &ManagedEnvironment, flox: &Flox) -> Result<Document, anyhow::Error> {
+    // in memory edit of the manifest
+    // toml_edit operates on &mut references of to this document
+    let mut doc = env
+        .manifest_content(flox)?
+        .parse::<Document>()
+        .expect("invalid doc");
+
+    // extract the `[options]` table
+    let options_table = doc
+        .entry("options")
+        .or_insert(toml_edit::Item::Table(toml_edit::Table::default()))
+        .as_table_like_mut()
+        .context("Invalid manifest format: expected [options] to be a table")?;
+
+    // extract the `options.systems` array
+    let systems_list = options_table
+        .entry("systems")
+        .or_insert(toml_edit::Item::Value(toml_edit::Value::Array(
+            toml_edit::Array::default(),
+        )))
+        .as_array_mut()
+        .context("Invalid manifest format: expected 'options.systems' to be an array")?;
+
+    // sanity check that the current system is not already in the list
+    if systems_list
+        .iter()
+        .any(|s| s.as_str().map(|s| s == flox.system).unwrap_or_default())
+    {
+        return Ok(doc);
```

Comment:
nonblocking suggestion: error, since we only call the function when we expect this is false

Reply:
There is no way for the user to reconcile though, i fdont know how we could get in that situation in the first place?

Reply:
Maybe at least `warn!()` so the user isn't confused/knows it's a bug?

Reply:
yeah logging it to debug now, since the function moved into the `flox_rust_sdk::manifest` module